iseq.c

```diff
@@ -1332,17 +1332,33 @@ rb_iseqw_to_iseq(VALUE iseqw)
 
 /*
  *  call-seq:
- *     iseq.eval -> obj
+ *     iseq.eval([binding]) -> obj
  *
  *  Evaluates the instruction sequence and returns the result.
  *
  *      RubyVM::InstructionSequence.compile("1 + 2").eval #=> 3
+ *
+ *  If <em>binding</em> is given, which must be a Binding object, the
+ *  evaluation is performed in its context.
+ *
+ *      obj = Struct.new(:a, :b).new(1, 2)
+ *      bind = obj.instance_eval {binding}
+ *      RubyVM::InstructionSequence.compile("a + b").eval(bind) #=> 3
  */
 static VALUE
-iseqw_eval(VALUE self)
+iseqw_eval(int argc, const VALUE *argv, VALUE self)
 {
-    rb_secure(1);
-    return rb_iseq_eval(iseqw_check(self));
+    VALUE scope;
+
+    if (argc == 0) {
```

Comment:
`Kernel#eval` accepts as `nil` as the binding, so this method too, I think.

Reply:
Yes I agree, I checked the `Kernel#eval` implementation, and i think that just evaluating as if no binding was passed is the best behavior, so I put it in the same if branch